### PR TITLE
test(message): migrate cypress tests to playwright

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour

- Copy `message` folder from `/cypress/locators/` to `/playwright/components/`
- Rename `index.js` to `index.ts` and `locators.js` to `locators.ts` in `/playwright/components/message/` folder
- Refactor `index.ts` to Playwright
- Copy `message.cy.tsx from `/cypress/components/message/` to `/src/components/message/`
- Rename `message.cy.tsx` to `message.pw.tsx` in `/src/components/message/` folder
- Refactor tests to Playwright

### Current behaviour

`Message` tests are in Cypress not Playwright.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run message tests in Playwright Test Runner to check if the `message.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/message/message.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+      await mount(<MessageComponent title={title} />);
+      await expect(messageTitle(page)).toHaveText(title);
+    });
+  });
+
+  ([
+    [true, "rgba(0, 0, 0, 0)"],
+    [false, "rgb(255, 255, 255)"],
+  ] as [boolean, string][]).forEach(([boolVal, backgroundColor]) => {
+    test(`should check ${boolVal} for transparent background for Message component`, async ({
```
Comment (DipperTheDan) [inline src/components/message/message.pw.tsx:122]: **suggestion:** this test description doesn't quite make sense to me when it's read out with the values. Something along the lines of:

```suggestion
    test(`should apply correct styling when transparent is ${boolVal}`, async ({
```

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/message/message.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+  [true, false].forEach((boolVal) => {
+    test(`should check when open is ${boolVal} for Message component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MessageComponent open={boolVal} />);
+      if (boolVal === true) {
+        await expect(messagePreview(page)).toBeVisible();
+      } else {
+        await expect(messagePreview(page).getAttribute("disabled")).toBeTruthy;
```
Comment (ZhuoyuJin) [inline src/components/message/message.pw.tsx:93]: suggest change "await expect(messagePreview(page)).not.toBeVisible();"

File: src/components/message/message.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+      await mount(<MessageComponent showCloseIcon={boolVal} />);
+
+      if (boolVal === true) {
+        await expect(messageDismissIcon(page)).toBeVisible();
+        await expect(messageContent(page)).toHaveCSS(
+          "padding",
+          `15px ${paddingVal}px 15px 20px`
+        );
+      } else {
+        await expect(messageDismissIcon(page).getAttribute("disabled"))
```
Comment (ZhuoyuJin) [inline src/components/message/message.pw.tsx:152]: Same suggestion as above, as the discussion about ".toBeTruthy" function is a bit flaky, would you be able to change it to ".not.toBeVisible();"

File: src/components/message/message.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+    await mount(<MessageComponentWithRef />);
+    await buttonPreview(page).click();
+    await expect(messagePreview(page)).toBeFocused();
+    await page.keyboard.press("Tab");
+    await expect(messageDismissIconButton(page)).toBeFocused();
+  });
+});
+
+test.describe("Accessibility tests for Message component", () => {
+  (["info", "error", "success", "warning"] as any[]).forEach((variant) => {
```
Comment (ZhuoyuJin) [inline src/components/message/message.pw.tsx:218]: type "any" can be replaced by the proper type for ["info", "error", "success", "warning"], which is MessageProps["variant"]

File: src/components/message/message.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    test(`should check ${title} as title for accessibility tests`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MessageComponent title={title} />);
+      await checkAccessibility(page);
+    });
+  });
+
+  ([
```
Comment (ZhuoyuJin) [inline src/components/message/message.pw.tsx:289]: "background" is declared here, but it never uses in the actual test.

File: src/components/message/message.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    }) => {
+      await mount(<MessageComponent transparent={boolVal} />);
+      await checkAccessibility(page);
+    });
+  });
+
+  ([
+    [true, 50],
+    [false, 20],
+  ] as [boolean, number][]).forEach(([boolVal, paddingVal]) => {
```
Comment (ZhuoyuJin) [inline src/components/message/message.pw.tsx:305]: same here, paddingVal is declared but never used in the actual test

File: src/components/message/message.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
+  test("should render message component for accessibility tests", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<MessageComponent />);
+
+    await checkAccessibility(page);
+  });
+});
```
Comment (ZhuoyuJin) [inline src/components/message/message.pw.tsx:333]: The original test has "onDismiss" prop checked accessibility in line 250-255 in "message.cy.tsx" file, could you add it in?

**Review by ZhuoyuJin — CHANGES_REQUESTED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**